qgis_hub_plugin/gui/resource_browser.py

```diff
@@ -174,30 +179,35 @@ def show_preview(self):
 
     def download_resource(self):
         resource = self.selected_resource()
-        file_path = download_resource_file(
-            resource.file, resource.resource_type, resource.uuid
-        )
+        file_extension = os.path.splitext(resource.file)[1]
+
+        file_path = QFileDialog.getSaveFileName(
+            self, "Save Resource", "", f"All Files (*{file_extension})"
+        )[0]
+
         if file_path:
-            text = f"Successfully download {resource.name} to {file_path}"
+            full_file_path = file_path + file_extension
+            download_resource_file(
+                resource.file, resource.resource_type, resource.uuid, full_file_path
+            )
+            text = "Successfully downloaded {} to {}".format(
+                resource.name, full_file_path
+            )
             self.iface.messageBar().pushMessage(self.tr("Success"), text, duration=5)
             if self.checkBoxOpenDirectory.isChecked():
-                QDesktopServices.openUrl(QUrl.fromLocalFile(str(file_path.parent)))
+                QDesktopServices.openUrl(
+                    QUrl.fromLocalFile(str(Path(file_path).parent))
+                )
         else:
-            text = f"Failed download {resource.name} from {resource.file}"
+            text = "Downloaded canceled for %s" % resource.name
             self.iface.messageBar().pushMessage(
                 self.tr("Warning"), text, level=Qgis.Warning, duration=5
             )
 
 
 # TODO: do it QGIS task to have
-def download_resource_file(url: str, resource_type: str, uuid: str):
-    qgis_user_dir = QgsApplication.qgisSettingsDirPath()
-    download_dir = Path(qgis_user_dir, "qgis_hub", "downloads", resource_type)
-    file_name = url.split("/")[-1]
-    resource_path = Path(download_dir, file_name)
-    if not download_dir.exists():
-        download_dir.mkdir(parents=True, exist_ok=True)
-
+def download_resource_file(url: str, resource_type: str, uuid: str, directory: str):
```

Comment:
You can also improve the existing code. In this case, you can remove the `uuid` parameter since it's not used and rename the parameter `directory` to `file_path` because it's not a directory anymore.